src/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/HashAlgorithm.cs

```diff
@@ -198,5 +203,39 @@ private void ValidateTransformBlock(byte[] inputBuffer, int inputOffset, int inp
         protected abstract void HashCore(byte[] array, int ibStart, int cbSize);
         protected abstract byte[] HashFinal();
         public abstract void Initialize();
+
+        protected virtual void HashCore(ReadOnlySpan<byte> source)
+        {
+            byte[] array = ArrayPool<byte>.Shared.Rent(source.Length);
+            try
+            {
+                source.CopyTo(array);
+                HashCore(array, 0, source.Length);
+            }
+            finally
+            {
+                Array.Clear(array, 0, source.Length);
+                ArrayPool<byte>.Shared.Return(array);
+            }
+        }
+
+        protected virtual bool TryHashFinal(Span<byte> destination, out int bytesWritten)
+        {
+            if (destination.Length >= HashSizeValue/8)
+            {
+                byte[] final = HashFinal();
+                if (final.Length != HashSizeValue/8)
```

Comment:
Save the hashSizeInBytes value to a variable?

Also, it feels like all the other methods were "if too short return false" and this is "if big enough return true".  Just seems odd that the style isn't consistent.